plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -122,6 +122,32 @@ export class TeamManager {
         }
     }
 
+    public async getTeamIdByToken(token: string): Promise<number | null> {
+        // TODO: avoid thundering herd on the DB for the same `token`
+        const cachedTeamId = this.tokenToTeamIdCache.get(token)
```

Comment:
Two nits:
- instead of duplicating, could we wrap around the existing `getTeamByToken` function instead? I'm worried that the two logics could diverge, with a risk of poisoning the shared LRU cache.
- `tokenToTeamIdCache` does not have a `maxAge` config, would it make sense to refresh it every 10 minutes, in order to handle key revocations without forcing a rolling-restart?

Reply:
Re. the `maxAge`, is it ok if I do not add this to this PR? I'm concerned about how much I've changed already!

Reply:
> Re. the maxAge, is it ok if I do not add this to this PR? I'm concerned about how much I've changed already!

Yes, of course, it can be a follow-up PR!